tests/Benchmarks/System.Text.Formatting/StringFormatterPerf.cs

```diff
@@ -0,0 +1,241 @@
+                sb.Append(((int)(i % 10)).ToString("X"));
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string InvariantFormatStruct()
+        {
+            var sb = new StringFormatter(NumbersToWrite * 2, pool);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(new Age(i % 10));
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string FormatGuid()
+        {
+            var guid = Guid.NewGuid();
+            var guidsToWrite = NumbersToWrite / 10;
+
+            var sb = new StringFormatter(guidsToWrite * 36, pool);
+            for (int i = 0; i < guidsToWrite; i++)
+            {
+                sb.Append(guid);
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string InvariantFormatStructClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite * 2);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(new Age(i % 10));
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string CustomCultureFormat()
+        {
+            var sb = new StringFormatter(NumbersToWrite * 3, pool);
+            sb.SymbolTable = _symbolTable;
+
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                var next = (i % 128) + 101;
+                sb.Append(next);
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public string CustomCultureFormatClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite * 3);
+            var culture = new CultureInfo("th");
+
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((i % 128) + 100).ToString(culture));
+            }
+
+            return sb.ToString();
+        }
+
+        [Benchmark]
+        public void EncodeStringToUtf8()
+        {
+            const string text = "Hello World!";
+            int stringsToWrite = 2000;
+            int size = stringsToWrite * text.Length + stringsToWrite;
+            var formatter = new ArrayFormatter(size, SymbolTable.InvariantUtf8, pool);
+
+            for (int i = 0; i < stringsToWrite; i++)
+            {
+                formatter.Append(text);
+                formatter.Append(1);
+            }
+        }
+
+        [Benchmark]
+        public void EncodeStringToUtf8Clr()
+        {
+            const string text = "Hello World!";
+            int stringsToWrite = 2000;
+            int size = stringsToWrite * text.Length + stringsToWrite;
+            var formatter = new StringBuilder(size);
+
+            for (int i = 0; i < stringsToWrite; i++)
+            {
+                formatter.Append(text);
+                formatter.Append(1);
+            }
+
+            var bytes = Encoding.UTF8.GetBytes(formatter.ToString());
```

Comment:
the `bytes` should be removed too

Reply:
Since there is nothing to return from `EncodeStringToUtf8Clr` should I just remove it instead of returning the string?

Reply:
@ahsonkhan should this benchmark include the cost of `Encoding.UTF8.GetBytes(formatter.ToString())`?

Reply:
Yes it should. The test that this is being compared to (`EncodeStringToUtf8`) encodes string data to utf8 as well while appending.